ms/MSOper/MSMetaData.cc

```diff
@@ -3546,6 +3553,345 @@ void MSMetaData::_computeScanAndSubScanProperties(
     }
 }
 
+#ifdef _OPENMP
+void MSMetaData::_computeScanAndSubScanPropertiesParallel(
+    SHARED_PTR<std::map<ScanKey, MSMetaData::ScanProperties> >& scanProps,
+    SHARED_PTR<std::map<SubScanKey, MSMetaData::SubScanProperties> >& subScanProps,
+    Bool showProgress
+) const {
+    uInt nrows = nRows();
+    SHARED_PTR<Vector<Int> > scans = _getScans();
+    SHARED_PTR<Vector<Int> > fields = _getFieldIDs();
+    SHARED_PTR<Vector<Int> > ddIDs = _getDataDescIDs();
+    SHARED_PTR<Vector<Int> > states = _getStateIDs();
+    SHARED_PTR<Vector<Double> > times = _getTimes();
+    SHARED_PTR<Vector<Int> > arrays = _getArrayIDs();
+    SHARED_PTR<Vector<Int> > observations = _getObservationIDs();
+    SHARED_PTR<Vector<Int> > ant1, ant2;
+    SHARED_PTR<QVD> exposureTimes = _getExposureTimes();
+    SHARED_PTR<QVD> intervalTimes = _getIntervals();
+    vector<uInt> ddIDToSpw = getDataDescIDToSpwMap();
+    _getAntennas(ant1, ant2);
+    uInt nthreads = min((uInt)1000, nrows);
+    uInt chunkSize = nrows/nthreads;
+    if (nrows % nthreads > 0) {
+        ++chunkSize;
+    }
+    _pmCount = 0;
+    SHARED_PTR<ProgressMeter> pm;
+    if (showProgress) {
+        LogIO log;
+        log << LogOrigin("MSMetaDataThreaded", __func__, WHERE)
+            << LogIO::NORMAL << "Compute subscan properties " << LogIO::POST;
+        pm.reset(new ProgressMeter(0, nrows, "Compute subscan info"));
+    }
+    pair<map<ScanKey, ScanProperties>, map<SubScanKey, SubScanProperties> > *fut =
+        new pair<map<ScanKey, ScanProperties>, map<SubScanKey, SubScanProperties> >[nthreads];
+#pragma omp parallel for
+    for (uInt i=0; i<nthreads; ++i) {
```

Comment:
I'm not following your objection. The loop is over the threads, not over rows. nthreads can be equal to nrows if nrows is small (<= 1000), but that's only for small data sets, in most cases nthreads, which maxes out at 1,000, will be significantly less than nrows. 

I don't see how the same row gets processed more than once. Consider the case where nrows = 1e7, for example. In that case nthreads = 1000, so chunkSize = 10,000. One then has the following relations:

i       begnRow(= i*chunkSize)      endRow(=std::min((i + 1) * chunkSize - 1, nrows - 1))
0             0                                    9999
1      10000                                  19999
2      20000                                  29999
...
999   9,990,000                         9,999,999
So each row is processed exactly once.